pkg/envoy/proxy.go

```diff
@@ -21,6 +21,19 @@ type Proxy struct {
 	lastSentVersion    map[TypeURI]uint64
 	lastAppliedVersion map[TypeURI]uint64
 	lastNonce          map[TypeURI]string
+
+	podUID, podNamespace, podIP, podServiceAccountName, envoyNodeID string
+}
+
+func (p *Proxy) SetMetadata(podUID, podNamespace, podIP, podServiceAccountName, envoyNodeID string) {
+	p.podUID = podUID
+	p.podNamespace = podNamespace
+	p.podIP = podIP
+	p.podServiceAccountName = podServiceAccountName
+	p.envoyNodeID = envoyNodeID
+
+	log.Trace().Msgf("Recorded metadata for Envoy %s: podUID=%s, podNamespace=%s, podIP=%s, podServiceAccountName=%s, envoyNodeID=%s",
```

Comment:
Would move this log outside of this API, and log it on the caller using `proxy.GetMetadata()`